server/lib/kyc/providers/base.ts

```diff
@@ -0,0 +1,55 @@
+import ActivityTypes from '../../../constants/activities';
+import Activity from '../../../models/Activity';
+import { KYCProviderName, KYCVerification, KYCVerificationStatus } from '../../../models/KYCVerification';
+
+export type KYCRequest = {
+  CollectiveId: number;
+  RequestedByCollectiveId: number;
+};
+
+export abstract class KYCProvider<
+  ProviderKYCRequest,
+  ProviderKYCVerification extends KYCVerification = KYCVerification,
+> {
+  providerName: KYCProviderName;
+
+  constructor(providerName: KYCProviderName) {
+    this.providerName = providerName;
+  }
+
+  abstract request(req: KYCRequest, providerRequest: ProviderKYCRequest): Promise<ProviderKYCVerification>;
+  abstract getVerifiedName(kycVerification: ProviderKYCVerification): string;
+  abstract getVerifiedAddress(kycVerification: ProviderKYCVerification): string;
```

Comment:
Perhaps having a unified `getVerifiedData` with typed returned data would be more future-proof. That way we wouldn't need to implement a getter for each new property we add, we could just update the type.